### PR TITLE
Refactored query "Pod is Not Targeted by a Proper Network Policy" to new version "Pod Misconfigured NetworkPolicy" #1975

### LINKED ISSUE #1975: Update Pod Is Not Targeted By A Proper Network Policy query for Kubernetes
### Platform
Kubernetes

### Description
Refactor Query 
The query has many false positives, this is because the query never detects if the Pod or the NetworkPolicy is defined.

### PR DESCRIPTION
**Pull request by joelcarvalhocheckmarx**
Closes #1975 

**Proposed Changes**

- Refactor Query
- The query has many false positives, this is because the query never detects if the Pod or the NetworkPolicy is defined.
- ~Solution: Finds NetworkPolicy and Pod, if exists should be targeted.~
- Solution: rewritten query to match specific scenario where no ingress and egress have been specified for a NetworkPolicy explicitly targeting a pod or in the same namespace as the pod

### CODE REVIEW

File: assets/queries/k8s/pod_network_policy_not_configured/query.rego [hunk 1]
```diff
@@ -4,58 +4,50 @@ CxPolicy[result] {
...
+	document.kind == "NetworkPolicy"
+	spec := document.spec
 
-	findNetworkPolicy(document.metadata.labels[key], key) == false
+	not findPod(true)
 
-	result := {
-		"documentId": input.document[i].id,
-		"searchKey": sprintf("metadata.name=%s.labels.%s", [metadata.name, key]),
+    result := {
```
Comment (joaomartinscx) [inline assets/queries/k8s/pod_network_policy_not_configured/query.rego:12]: Fix indentation

File: assets/queries/k8s/pod_network_policy_not_configured/query.rego [hunk 2]
```diff
@@ -4,58 +4,50 @@ CxPolicy[result] {
...
+		"documentId": document.id,
+		"searchKey": sprintf("metadata.name=%s", [metadata.name]),
 		"issueType": "MissingAttribute",
-		"keyExpectedValue": sprintf("'metadata.labels.%s' is targeted by a network policy", [key]),
-		"keyActualValue": sprintf("'metadata.labels.%s' is not targeted by a network policy", [key]),
+		"keyExpectedValue": "Pod is defined",
```
Comment (joaomartinscx) [inline assets/queries/k8s/pod_network_policy_not_configured/query.rego:16]: Try to be more clear on the expected value e.g. the deleted expectedValue

File: assets/queries/k8s/pod_network_policy_not_configured/query.rego [hunk 3]
```diff
@@ -4,58 +4,50 @@ CxPolicy[result] {
...
+		"keyActualValue": sprintf("metadata.name=%s should target at least one Pod", [metadata.name])
```
Comment (joaomartinscx) [inline assets/queries/k8s/pod_network_policy_not_configured/query.rego:17]: The ActualValue mentions what values the resource has that are faulty, so the message should reflect that, and not what there should be

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

**Review by joaomartinscx — APPROVED** [review]
Everything looks good to me, good job!

**Comment by rogeriopeixotocx** [comment]
Sample coverage is not good yet:

image

we need to be mindful of performance and integration tests issues when iterating over all documents, lines:
https://github.com/Checkmarx/kics/pull/1976/files#diff-e0f0779359c491ebbe5def5c81a7cfa599340fedb4c8584e8e6c6a209a564fe3R47
https://github.com/Checkmarx/kics/pull/1976/files#diff-e0f0779359c491ebbe5def5c81a7cfa599340fedb4c8584e8e6c6a209a564fe3R42

I'll suggest an improvment.

cc: @joelcarvalhocheckmarx @ruigomescx

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**